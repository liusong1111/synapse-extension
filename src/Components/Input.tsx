import * as React from 'react';

interface AppProps { }

interface AppState { }

export default class Input extends React.Component<AppProps, AppState> {
  constructor(props: AppProps, state: AppState) {
    super(props, state);
  }

  componentDidMount() {
    // Example of how to send a message to eventPage.ts.
    // chrome.runtime.sendMessage({ popupMounted: true });
  }

  render() {
    return (
      <div className="container">
        <input />
      </div>
    )
  }
}
