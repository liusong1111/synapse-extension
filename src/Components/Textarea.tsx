import * as React from "react";

interface AppProps {
  onChange: any;
}

interface AppState {}

export default class extends React.Component<AppProps, AppState> {
  constructor(props: AppProps, state: AppState) {
    super(props, state);
  }

  render() {
    return (
      <div>
        <textarea onChange={this.props.onChange} />
      </div>
    )
  }
}