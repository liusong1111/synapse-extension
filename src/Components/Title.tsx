import * as React from "react";

interface AppProps {
  title: string
}

interface AppState {}

export default class extends React.Component<AppProps, AppState> {
  constructor(props: AppProps, state: AppState) {
    super(props, state);
  }

  render() {
    return (
      <h3>
        {this.props.title}
      </h3>
    )
  }
}