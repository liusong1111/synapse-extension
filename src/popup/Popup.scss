$primary-color: #00008b; // darkblue
$background-color: #BADA55;

// Chrome applies an 8px margin by default (gross).
body {
  font-family: Roboto, Arial;
  color: #4d4d4d;
  font-weight: 400;
  background: #f7f7f7;
  width: 100%;
  height: 100%;
  margin: 0;
  padding: 0;
  font-size: 16px;
}