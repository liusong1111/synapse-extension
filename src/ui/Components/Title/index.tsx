import React from 'react';

interface AppProps {
  title: string;
  testId: string;
}

interface AppState {}

export default class extends React.Component<AppProps, AppState> {
  constructor(props: AppProps, state: AppState) {
    super(props, state);
  }

  render() {
    return <h3 data-testid={this.props.testId || 'title'}>{this.props.title}</h3>;
  }
}
