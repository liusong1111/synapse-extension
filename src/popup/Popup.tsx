import * as React from 'react';
// import './Popup.scss';
import Input from '../Components/Input'
import Title from '../Components/Title'
import Textarea from '../Components/Textarea'
import { Button, TextField } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { Formik, Form } from "formik";
import * as Yup from "yup";

const useStyles = makeStyles({
  container: {
    height: 600,
    width: 357,
    minHeight: 500,
  },
  button: {
    background: 'linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)',
    border: 0,
    borderRadius: 3,
    boxShadow: '0 3px 5px 2px rgba(255, 105, 135, .3)',
    color: 'white',
    height: 48,
    padding: '0 30px',
  },
  textField: {

  }
});


interface AppProps { }

interface AppState { }

export const innerForm = props => {
  const classes = useStyles();
  const {
    values,
    touched,
    errors,
    dirty,
    isSubmitting,
    handleChange,
    handleBlur,
    handleSubmit,
    handleReset
  } = props;
  return (
    <Form className="form-mnemonic" onSubmit={handleSubmit}>
      <TextField
        label="Mnemonic"
        name="mnemonic"
        multiline
        className={classes.textField}
        value={values.mnemonic}
        onChange={handleChange}
        onBlur={handleBlur}
        helperText={(errors.mnemonic && touched.mnemonic) && errors.mnemonic}
        margin="normal"
        data-test="mnemonic"
      />
      <TextField
        label="Password"
        name="password"
        className={classes.textField}
        value={values.password}
        onChange={handleChange}
        onBlur={handleBlur}
        helperText={(errors.password && touched.password) && errors.password}
        margin="normal"
      />
      <TextField
        label="Confirm Password"
        name="confirmPassword"
        className={classes.textField}
        value={values.confirmPassword}
        onChange={handleChange}
        onBlur={handleBlur}
        helperText={(errors.confirmPassword && touched.confirmPassword) && errors.confirmPassword}
        margin="normal"
      />
      <Button type="submit" disabled={isSubmitting}>
        Import
      </Button>

    </Form>
  );
}

export default function (props: AppProps, state: AppState) {
  const [success, setSuccess] = React.useState(false)
  const [value, setValue] = React.useState('Controlled');

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setValue(event.target.value);
    console.log(event.target.value, '======')
  };

  function onSubmit(e) {
    e.preventDefault()
    setSuccess(true)
    // chrome.runtime.sendMessage('abc')
    console.log(e, '---=====')
  }

  let successNode = null
  if (success) successNode = <div className="success">Successfully</div>

  const classes = useStyles();
  return (
    <div className={classes.container}>
      {successNode}
      <Title title='Import Mnemonic' />

      <Formik
        initialValues={{ mnemonic: "", password: "", confirmPassword: "" }}
        onSubmit={async values => {
          await new Promise(resolve => setTimeout(resolve, 500));
          console.log(JSON.stringify(values, null, 2));
          setSuccess(true)
        }}
        validationSchema={Yup.object().shape({
          mnemonic: Yup.string()
            .required("Required"),
          password: Yup.string()
            .required("Required"),
          confirmPassword: Yup.string()
            .oneOf([Yup.ref('password'), null], "Passwords don't match!")
            .required("Required")
        })}
      >
        {innerForm}
      </Formik>
    </div>
  )
}
