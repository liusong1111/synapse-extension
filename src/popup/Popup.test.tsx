import * as React from 'react';
import { shallow } from 'enzyme';
import Popup from './Popup';
import Input from '../Components/Input';
import Title from '../Components/Title';
import Textarea from '../Components/Textarea';
import { Button } from '@material-ui/core';

describe('Popup', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<Popup />));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should render title', () => {
    expect(wrapper.find(Title)).toHaveLength(1);
    // expect(wrapper.containsMatchingElement(<Title title={'Import Mnemonic'} />)).toEqual(true)
  })

  it('should render 1 Textarea', () => {
    // expect(wrapper.containsMatchingElement(<Textarea onChange={wrapper.instance.onChange} />)).toEqual(true);
    expect(wrapper.find(Textarea)).toHaveLength(1);
  });

  it('should render 2 Input', () => {
    expect(wrapper.find(Input)).toHaveLength(2);
  });

  it('should render a <Button />', () => {
    expect(wrapper.find(Button)).toHaveLength(1);
  });

  it('should render button text', () => {
    expect(wrapper.find(Button).text()).toEqual('Import');
  });

  it('simulates click events', () => {
    // const onButtonClick = sinon.spy();
    // const wrapper = shallow(<Foo onButtonClick={onButtonClick} />);
    // wrapper.find('button').simulate('click');
    // expect(onButtonClick).to.have.property('callCount', 1);
  });
});