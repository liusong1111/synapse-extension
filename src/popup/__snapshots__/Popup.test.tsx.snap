// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup should render correctly 1`] = `
<div
  className="popupContainer"
>
  <default_1
    title="Import Mnemonic"
  />
  <default_1
    onChange={[Function]}
  />
  <Input />
  <Input />
  <WithStyles(ForwardRef(Button))
    className="makeStyles-root-1"
    color="primary"
  >
    Import
  </WithStyles(ForwardRef(Button))>
</div>
`;
